src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesReconService.kt

```diff
@@ -69,19 +69,27 @@ class ActivitiesReconService(
     )
   }
 
-  private fun publishTelemetry(
+  private suspend fun publishTelemetry(
     prisonId: String,
     compareResults: CompareBookingsResult,
   ) {
     val telemetryName =
       "activity-allocation-reconciliation-report-${if (compareResults.noDifferences()) "success" else "failed"}"
-    val telemetryMap = mutableMapOf("prison" to prisonId)
-
-    if (compareResults.nomisOnly.isNotEmpty()) telemetryMap["NOMIS_only"] = compareResults.nomisOnly.toString()
-    if (compareResults.dpsOnly.isNotEmpty()) telemetryMap["DPS_only"] = compareResults.dpsOnly.toString()
-    if (compareResults.differentCount.isNotEmpty()) telemetryMap["different_count"] = compareResults.differentCount.toString()
+    if (compareResults.noDifferences()) {
+      telemetryClient.trackEvent(telemetryName, mapOf("prison" to prisonId))
+    }
```

Comment:
I find the lack of an `else` here odd, why call `getPrisonerDetails` for an empty list / set?